File: DSA/Searching/Binary/Python/binrecur.py
def binary(arr,s,l,target):
    if(s>l):
        return -1
    elif len(arr)==0:
        return -1
    mid=s+(l-s)//2
    if target==arr[mid]:
        return mid
    elif target>arr[mid]:
        return binary(arr,mid+1,l,target)
    else:
        return binary(arr,s,mid-1,target) 
    return -1

if __name__=="__main__":
    inp=list(map(int,input("Enter the Elements : ")))
    target=int(input("Enter the Target : "))
    print(binary(inp,0,len(inp),target))
    

File: DSA/Searching/Binary/C/binrecur.c
#include<stdio.h>
int binary(int arr[],int num,int start,int end,int target){
    
    if (start>end || num==0)
        return -1;
    
    int mid=start+((end-start)/2);

    if (arr[mid]==target)
        return mid;
    
    else if(arr[mid]<target)
        return binary(arr,num,mid+1,end,target);
    
    else
        return binary(arr,num,start,mid-1,target);
    

}
int main(){
    int num;
    printf("Enter the Length: ");
    scanf("%d",&num);
    int arr[num];
    printf("\nEnter The Elements : ");
    for (int i = 0; i < num; i++)
    {
        scanf("%d",&arr[i]);
    }
    int target;
    printf("\nEnter the Target: ");
    scanf("%d",&target);
    int ans=binary(arr,num,0,num-1,target);
    printf("%d",ans);

    return 0;
}